GCS does manual params validation

vadmeste opened the pull request:
## Description
To control the order of validation, gcs gateway checks project id
then access/secret keys.

```
$ minio gateway gcs
FATA[0000] Unable to initialize gateway layer            cause=GCS project ID expected source=[gateway-main.go:222:gatewayMain()]
```

## Motivation and Context
Fixed #4396 

## How Has This Been Tested?
go test + Manual

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

harshavardhana commented on cmd/gateway-gcs.go:
```diff
@@ -183,6 +185,10 @@ func newGCSGateway(args cli.Args) (GatewayLayer, error) {
 	}
 
+	if os.Getenv("MINIO_ACCESS_KEY") == "" || os.Getenv("MINIO_SECRET_KEY") == "" {
+		return nil, errors.New("No GCS credentials set")
```
`No credentials set` is correct instead of `No GCS credentials set`

vadmeste replied:
oups yes.

vadmeste replied:
> No credentials set is correct instead of No GCS credentials set

Done

krishnasrinivas reviewed (approved):
Can you rebase @vadmeste

vadmeste committed:
gcs: Do manual params validation

To control the order of validation, each gateway checks for
access/secret keys by itself and generates a gateway specific
error when needed.
